src/explore-education-statistics-admin/src/pages/EditReleasePage.tsx

```diff
@@ -0,0 +1,119 @@
+import DummyPublicationsData from '@admin/pages/DummyPublicationsData';
+import EditReleaseSetupSummary from '@admin/components/EditReleaseSetupSummary';
+import NavigableSections, {
+  Section,
+} from '@admin/components/NavigableSections';
+import { RouteComponentProps } from 'react-router';
+
+export enum ReleaseSection {
+  ReleaseSetup,
+  AddEditData,
+  BuildTables,
+  ViewEditTables,
+  AddEditContent,
+  SetPublishStatus,
+}
+
+interface Props extends RouteComponentProps {
+  releaseId: string;
+}
+
+const navigationHeadings: (
+  releaseId: string,
+) => Section<ReleaseSection>[] = releaseId => {
+  const urlPrefix = `/edit-release/${releaseId}`;
+  return [
+    {
+      section: ReleaseSection.ReleaseSetup,
+      label: 'Release setup',
+      linkTo: `${urlPrefix}/setup`,
+    },
+    {
+      section: ReleaseSection.AddEditData,
+      label: 'Add / edit data',
+      linkTo: `${urlPrefix}/data`,
+    },
+    {
+      section: ReleaseSection.BuildTables,
+      label: 'Build tables',
+      linkTo: `${urlPrefix}/build-tables`,
+    },
+    {
+      section: ReleaseSection.ViewEditTables,
+      label: 'View / edit tables',
+      linkTo: `${urlPrefix}/tables`,
+    },
+    {
+      section: ReleaseSection.AddEditContent,
+      label: 'Add / edit content',
+      linkTo: `${urlPrefix}/content`,
+    },
+    {
+      section: ReleaseSection.SetPublishStatus,
+      label: 'Set publish status',
+      linkTo: `${urlPrefix}/publish-status`,
+    },
+  ];
+};
+
+const breadcrumbs = [
+  {
+    link: '/admin-dashboard',
+    name: 'Administrator dashboard',
+  },
+  { name: 'Edit release', link: '#' },
+];
+
+const EditReleasePage = ({ releaseId, location }: Props) => {
+  const [release, setRelease] = useState<Release>();
+
+  const [publicationTitle, setPublicationTitle] = useState<string>();
+
+  useEffect(() => {
+    const selectedRelease = DummyPublicationsData.getReleaseById(releaseId);
+    const owningPublication = DummyPublicationsData.getOwningPublicationForRelease(
+      selectedRelease,
+    );
+    setRelease(selectedRelease);
+    setPublicationTitle(owningPublication ? owningPublication.title : '');
+  }, [releaseId]);
+
+  const availableSections = navigationHeadings(releaseId);
+
+  const selectedSection =
+    availableSections.find(section =>
+      location.pathname.endsWith(section.linkTo),
+    ) || availableSections[0];
+
+  return (
+    <Page wide breadcrumbs={breadcrumbs}>
+      {release && publicationTitle && (
+        <>
+          <h1 className="govuk-heading-l">
+            {publicationTitle}
+            <span className="govuk-caption-l">Edit release</span>
+          </h1>
+
+          <NavigableSections
+            navigationHeadingText={publicationTitle}
+            navigationHeadingSubtitle="Edit release"
+            availableSections={availableSections}
+            selectedSection={selectedSection}
+          >
+            {ReleaseSection.ReleaseSetup === selectedSection.section && (
```

Comment:
Not particularly keen on this as it feels very switch statement-y and it feels like the abstraction is basically just reinventing the routing wheel. 

Why not create additional routes/pages for each of the various sections?

Reply:
Ummed and ahhed about this one and yeah, I'm certainly not going to go further with this pattern.  Don't particularly want to create separate pages for each section either as there's a lot of common features amongst them, although I'm sure there's ways to reduce the duplication.  I initially tried to use Routes within this file to do conditional rendering but again, wasn't sure if this was a good convention or not.  

What would be your tactic?

Reply:
Yeh I would have suggested to go with conditional route rendering as that's what the API is designed for. In combination with `NavLink` described elsewhere, we can clean this set of components up a bit.

A more concrete example can be found here: https://reacttraining.com/react-router/web/example/sidebar

Reply:
Done done done, thanks for the info